packages/py-api/py_api/controllers/uploader_controller.py

```diff
@@ -69,3 +75,24 @@ def dump_objects(cls) -> Dict[str, Any]:
             )
 
         return {'objects': objects}
+
+    @classmethod
+    def delete_object_by_filename(cls, filename: str) -> Dict[str, Any]:
+        # Check whether the image exists
+        try:
+            cls._s3_client.head_object(
+                Bucket=cls._AWS_BUCKET_NAME, Key=filename,
+            )
+        except cls._s3_client.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == '404':
+                raise HTTPException(
+                    status_code=404, detail=f"Object '{filename}' not found in the storage",
+                )
+
+            raise HTTPException(
+                status_code=500, detail=f"Internal error!",
```

Comment:
```
                status_code=500, detail=str(e),
```

Reply:
@asynchroza shouldn't  we return a JSONResponse. I know that the result is the same, but we catch an exception and raise another one, which is something I don't like. Also in the other controllers we return a JSONResponse.

Reply:
Well you should get used to it. This is a pretty common pattern.

Reply:
The exception handling I mean. Otherwise yeah you could return a jsonresponse here.